modules/user_data/variables.tf

```
@@ -26,103 +41,179 @@ variable "no_proxy" {
 ## Base TFE Configs
 
 variable "capacity_concurrency" {
-  default = "10"
+  default     = "10"
+  description = "The maximum number of Terraform runs that will be executed concurrently on each compute instance."
+  type        = string
 }
 
 variable "capacity_memory" {
-  default = "512"
+  default     = "512"
+  description = <<-EOD
+  The maximum amount of memory that will be allocated to each Terraform run. The value must be expressed in megabytes.
+  EOD
+  type        = string
 }
 
 variable "ca_certs" {
-  default = ""
+  default     = ""
+  description = <<-EOD
+  A custom Certificate Authority certificate bundle to be used for authenticating connections with Terraform Enterprise.
+  EOD
+  type        = string
 }
 
 variable "custom_image_tag" {
-  default = "hashicorp/build-worker:now"
+  default     = "hashicorp/build-worker:now"
+  description = "The tag of the Docker image to be used as the custom Terraform Build Worker image."
+  type        = string
 }
 
 variable "disk_path" {
-  default = "/opt/hashicorp/data"
+  default     = "/opt/hashicorp/data"
+  description = "The pathname of the directory in which Terraform Enterprise will store data on the compute instances."
+  type        = string
 }
 
 variable "enable_metrics_collection" {
-  default = "1"
+  default     = "1"
+  description = <<-EOD
+  A toggle to control the collection of metrics from Terraform Enterprise. The value must be \"1\" for
+  true and \"0\" for false.
+  EOD
+  type        = string
+  validation {
+    condition     = contains(["1", "0"], var.enable_metrics_collection)
+    error_message = "The enable_metrics_collection value must be \"1\" for true and \"0\" for false."
+  }
 }
 
 variable "extra_no_proxy" {
-  default = ""
+  default     = ""
+  description = <<-EOD
+  A list of hosts for which Terraform Enterprise will not use a proxy to access. The list must be a comma-separated
+  string, like \".example.com,.example.org\".
+  EOD
+  type        = string
 }
 
 variable "hairpin_addressing" {
-  default = false
-  type    = bool
+  default     = false
+  description = "A toggle to control the use of hairpin addressing within the TFE deployment."
+  type        = bool
 }
 
 variable "iact_subnet_list" {
-  default = ""
+  default     = ""
+  description = "A list of IP address ranges which will be authorized to access the IACT. The ranges must be expressed in CIDR format and the list must be a comma-separated string, like \"10.0.1.0/24,172.16.4.0/24\"."
+  type        = string
```

Not super important, but if you want...
```suggestion
  description = <<-EOD
  A list of IP address ranges which will be authorized to access the IACT. The ranges must be expressed
  in CIDR format and the list must be a comma-separated string, like "10.0.1.0/24,172.16.4.0/24".
  EOD
  type        = string
```